actions/user_actions.test.js

```diff
@@ -232,4 +241,62 @@ describe('Actions.User', () => {
         await testStore.dispatch(UserActions.loadProfilesForGroupChannels(mockedGroupChannels));
         expect(testStore.getActions()).toEqual(expectedActions);
     });
+
+    test('Should call p-queue APIs on loadProfilesForGM', async () => {
+        const gmChannel = {id: 'gmChannel', type: General.GM_CHANNEL, team_id: ''};
+        UserActions.queue.add = jest.fn().mockReturnValue(jest.fn());
+        UserActions.queue.onEmpty = jest.fn();
+
+        const user = TestHelper.fakeUser();
+
+        const profiles = {
+            [user.id]: user,
+        };
+
+        const channels = {
+            [gmChannel.id]: gmChannel,
+        };
+
+        const channelsInTeam = {
+            '': [gmChannel.id],
+        };
+
+        const myMembers = {
+            [gmChannel.id]: {},
+        };
+
+        const state = {
+            entities: {
+                users: {
+                    currentUserId: 'current_user_id',
+                    profiles,
+                    statuses: {},
+                    profilesInChannel: {
+                        [gmChannel.id]: new Set(['current_user_id']),
+                    },
+                },
+                teams: {
+                    currentTeamId: 'team_1',
+                },
+                channels: {
+                    channels,
+                    channelsInTeam,
+                    myMembers,
+                },
+                preferences: {
+                    myPreferences: {},
+                },
+                general: {
+                    config: {},
+                },
+            },
+        };
+
+        const testStore = mockStore(state);
+        getState.mockImplementation(testStore.getState);
+
+        await UserActions.loadProfilesForGM();
+        expect(UserActions.queue.onEmpty).toHaveBeenCalled();
+        expect(UserActions.queue.add).toHaveBeenCalled();
```

Comment:
i wanted to assert that we are calling an action here but it was becoming hard to mock the class etc.